File: 프로그래머스/lv2/42586. 기능개발/기능개발.py
import math
from collections import deque

def solution(progresses, speeds):
    answer = []
    stan = 0 # 가장 앞의 남은 작업량
    cnt = 0 # 쌓인 배포
    
    for p in range(len(progresses)):
        # progresses[p] = math.ceil((100 - progresses[p])/speeds[p])
        tmp = math.ceil((100 - progresses[p])/speeds[p])
        if stan == 0:
            stan = tmp
            cnt += 1
        elif stan < tmp:
            answer.append(cnt)
            stan = tmp
            cnt = 1
        else:
            cnt += 1
        if p == len(progresses)-1:
            answer.append(cnt)
        
    
    # pro = deque(progresses)
    
    
    return answer

File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;

class Solution {
    public List<Integer> solution(int[] progresses, int[] speeds) {
        List<Integer> answer = new ArrayList<>();
        Stack<Integer> st = new Stack<>();
        int[] l = new int[progresses.length];
        // int cnt = 0;
        int pre = 0;
        for(int i=0;i<progresses.length;i++){
            int tmp = (100-progresses[i])/speeds[i];
            if((100-progresses[i])%speeds[i]!=0) tmp++;
            if(st.isEmpty()) {
                st.push(tmp);
                pre = tmp;
            }else if(pre >= tmp){
                st.push(tmp);
            }else{
                answer.add(st.size());
                st.clear();
                st.push(tmp);
                pre = tmp;
            }
        }
        answer.add(st.size());
        return answer;
    }
}